ormin/queries.nim

```diff
@@ -436,13 +436,13 @@ proc generateRoutine(name: NimNode, q: QueryBuilder;
   if k != nnkIteratorDef:
     yld.add newVarStmt(ident"res", newCall(bindSym"createJObject"))
   let fn = if q.retTypeIsJson: bindSym"bindResultJson" else: bindSym"bindResult"
-  if q.retType.len > 1:
```

Comment:
But single items shouldn't get a tuple return type.

Reply:
Aha, this seemed like it was more of an accident than a deliberate design choice. Take this code for example:

``` Nim
# import model and opening of DB not included
createIter allUsernames:
  select users(name)
  #produce json
for username in db.allUsernames:
  echo username
```
This iterator only gives strings, but comment in the `#produce json` part and now you have a JSON object with a `"name"` field. The reason Ormin doesn't create a tuple is because while `var res: (string, int)` will create an unnamed tuple with a string and an int, `var res: (string)` will only create a string. But it's an easy fix to get it to return only values though, but maybe it should also be done for JSON types?

Reply:
If I requested JSON, I wanted JSON, but tuples don't have to be "consistent" with it, and shouldn't.

Reply:
Well yes, but a `JsonNode` with `kind == JString` is still JSON.